addons/website_sale_comparison/models/website_sale_comparison.py

```diff
@@ -66,15 +66,14 @@ def _prepare_categories_for_display(self):
             }]
         }]
         """
-        attributes = self.product_tmpl_id.valid_product_template_attribute_line_ids._without_no_variant_attributes().attribute_id.sorted()
+        attributes = self.product_tmpl_id.valid_product_template_attribute_line_ids.attribute_id.sorted()
         categories = OrderedDict([(cat, OrderedDict()) for cat in attributes.category_id.sorted()])
         if any(not pa.category_id for pa in attributes):
             # category_id is not required and the mapped does not return empty
             categories[self.env['product.attribute.category']] = OrderedDict()
         for pa in attributes:
             categories[pa.category_id][pa] = OrderedDict([(
                 product,
-                product.product_template_attribute_value_ids.filtered(lambda ptav: ptav.attribute_id == pa)
+                product.attribute_line_ids.filtered(lambda ptav: ptav.attribute_id == pa).value_ids
```

Comment:
@chevalierv  not 100% sure about this change but:
`self = product.product`
`self.product_template_attribute_value_ids` contains only attrs with creation mode = never, so we missed all the other attributes so thats why we get empty cells.
from what i see we get all the attributes in attribute_line_ids (coming from product.template) 

before:
![image](https://github.com/odoo/odoo/assets/59824990/435194a5-c14f-432a-97b0-d6c0ed77f776)

after:
![image](https://github.com/odoo/odoo/assets/59824990/86ee1506-d4fb-4b1d-aae9-68258894cf38)